analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Sonar/Analyzers/InvalidCastToInterfaceSymbolicExecution.cs

```diff
@@ -19,17 +19,15 @@
  */
 
 using SonarAnalyzer.SymbolicExecution.Constraints;
+using SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks.CSharp;
 
 namespace SonarAnalyzer.SymbolicExecution.Sonar.Analyzers
 {
     internal sealed class InvalidCastToInterfaceSymbolicExecution : ISymbolicExecutionAnalyzer
     {
-        private const string DiagnosticId = "S1944";
         private const string MessageDefinite = "Nullable is known to be empty, this cast throws an exception.";
 
-        internal static readonly DiagnosticDescriptor S1944 = DescriptorFactory.Create(DiagnosticId, "{0}");
-
-        public IEnumerable<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(S1944);
+        public IEnumerable<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(InvalidCastToInterface.S1944);  // References the new SE because of the shared non-SE rule part
```

Comment:
Yes. For other SE rules, we don't do this, see
https://github.com/SonarSource/sonar-dotnet/blob/9d5557178d87bcfe86fd054027d0ad4d83c110de/analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Sonar/Analyzers/NullPointerDereference.cs#L31-L33
vs
https://github.com/SonarSource/sonar-dotnet/blob/9d5557178d87bcfe86fd054027d0ad4d83c110de/analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/NullPointerDereference.cs#L27-L29


Here I had to do it, because UTs and `WithOnlyDiagnostics` would disqualify half of the rule on one or the other side.